ui/components/src/ui-components/filtering/FilteringInput.tsx

```diff
@@ -111,27 +157,61 @@ export class FilteringInput extends React.PureComponent<FilteringInputProps, Fil
       return;
 
     this.props.onFilterStart(this.state.searchText);
+    this.setState({ searchStarted: true });
     e.stopPropagation();
   };
 
   private _onInputChanged = (e: React.ChangeEvent<HTMLInputElement>) => {
-    this.setState({ searchText: e.target.value, context: InputContext.ReadyToFilter });
-  };
+    if(this.state.searchStarted){
+      this.props.onFilterCancel();
+    }
+    this.setState({ searchText: e.target.value, searchStarted: false});
+  }
+
 
   /** @internal */
-  public static getDerivedStateFromProps(props: FilteringInputProps, state: FilteringInputState) {
-    if (state.context === InputContext.FilteringInProgress && !props.filteringInProgress) {
-      if (state.searchText && props.resultSelectorProps)
-        return { context: InputContext.FilteringFinished };
-      else
-        return { context: InputContext.FilteringFinishedWithNoStepping };
-    } else if (state.context === InputContext.ReadyToFilter && props.filteringInProgress) {
-      return { context: InputContext.FilteringInProgress };
+  public componentDidUpdate(prevProps: FilteringInputProps, _prevState: FilteringInputState) {
+    if (FilteringInput.isNewProps(this.props) && this.props.resultSelectorProps !== prevProps.resultSelectorProps) {
+      this.setState((state, _props) => ({ resultSelectorKey: state.resultSelectorKey + 1 }));
+    }
+    if(FilteringInput.isOldProps(prevProps) && FilteringInput.isOldProps(this.props) && prevProps.filteringInProgress !== this.props.filteringInProgress){
+      this.setState((_state, props) => (FilteringInput.isOldProps(props)?{ filteringInProgress: props.filteringInProgress } : {filteringInProgress: undefined}));
+    }
+  }
+
+  public static getDerivedStateFromProps(nextProps: FilteringInputProps, prevState: FilteringInputState){
+    if (FilteringInput.isOldProps(nextProps) && !nextProps.filteringInProgress && prevState.filteringInProgress){
+      return { searchStarted: true};
     }
     return null;
   }
 
+  private static isNewProps(props: FilteringInputProps): props is NEW_FilteringInputProps {
```

Comment:
I would suggest removing `isNewProps` and renaming `isOldProps` to `isDeprecatedProps`. Then, whenever we need to distinguish between the two, the code should look like this:
```ts
if (isDeprecatedProps(props)) {
  // do stuff
  return;
}
// use new props
```
This will make removing the deprecated format much easier

Reply:
Done that